### DISCUSSION TITLE
Clarifying Creating Reagent Components for Subscriptions

### DISCUSSION THREAD

**Discussion by whamtet** [day8/re-frame#688] [Q&A]
Thanks for the great page on [Creating Reagent Components](https://github.com/Day8/re-frame/wiki/Creating-Reagent-Components)

One issue that I'm not clear on is whether a second order function is necessary when dereferencing a subscription.  For example

```clojure
(defn my-component []
  (let [my-val (subscribe [:my-sub])]
    [:div @my-val]))

(defn my-component-second-order []
  (let [my-val (subscribe [:my-sub])]
    (fn []
      [:div @my-val])))
```

I'm guessing the first form is ok because the atom is not actually created in the `let` block, but it would be good to clarify in the doco.  I'm happy to do a doco PR if you'd like.

**Comment by danielcompton** [comment]
Hi @whamtet form-1 reagent components are fine now for using re-frame subscriptions, https://github.com/Day8/re-frame/issues/218 and https://lambdaisland.com/blog/11-02-2017-re-frame-form-1-subscriptions have more details on this. I thought we had some docs on this somewhere, but I had a quick look and couldn't come up with anything. Happy to take a PR on this, but I'm not really sure where it should go. @mike-thompson-day8 is more clued up with the docs, but he's unavailable for the next few weeks. So, happy to take a PR but it might be a while before it gets reviewed and merged.

**Comment by mike-thompson-day8** [comment]
I'm closing this. I think most of the re-frame docs  and examples clear.
